mne/io/fieldtrip.py

```diff
@@ -0,0 +1,444 @@
+                         ignore_fields=['previous'],
+                         variable_names=[data_name])
+
+    # load data and set ft_struct to the heading dictionary
+    ft_struct = ft_struct[data_name]
+
+    data = np.array(ft_struct['trial'])  # create the main data array
+    info = _create_info(ft_struct)  # create info structure
+
+    if data.ndim > 2:
+        data = np.squeeze(data)
+
+    if data.ndim == 1:
+        data = data[np.newaxis, ...]
+
+    if data.ndim != 2:
+        raise RuntimeError('The data you are trying to load does not seem to'
+                           'be raw data')
+
+    custom_raw = RawArray(data, info)  # create an MNE RawArray
+    return custom_raw
+
+
+def read_epochs_fieldtrip(ft_structure_path, data_name='data',
+                          trialinfo_column=0):
+    """Load epoched data from a FieldTrip preprocessing structure.
+
+    This function expects to find epoched data in the structure data_name is
+    pointing at.
+
+    .. warning:: Only epochs with the same amount of channels and samples are
+                 supported!
+
+    .. warning:: FieldTrip stores all channel locations in head coordinates.
+                 mne-python stores MEG channel locations in device coordinates
+                 by default. This is not possible because the transformation
+                 matrix is missing. All channel locations are thus stored in
+                 head coordinates.
+
+    Parameters
+    ----------
+    ft_structure_path: str
+        Path and filename of the .mat file containing the data.
+    data_name: str
+        Name of heading dict/ variable name under which the data was originally
+        saved in MATLAB.
+    trialinfo_column: int
+        Column of the trialinfo matrix to use for the event codes
+
+
+    Returns
+    -------
+    mne.EpochsArray
+        A MNE EpochsArray structure consisting of the epochs arrays, an event
+        matrix, start time before event (if possible, else defaults to 0) and
+        measurement info.
+
+
+    """
+    from ..externals.pymatreader.pymatreader import read_mat
+    ft_struct = read_mat(ft_structure_path,
+                         ignore_fields=['previous'],
+                         variable_names=[data_name])
+
+    # load data and set ft_struct to the heading dictionary
+    ft_struct = ft_struct[data_name]
+
+    data = np.array(ft_struct['trial'])  # create the epochs data array
+    events = _create_events(ft_struct, trialinfo_column)
+    metadata = _create_event_metadata(ft_struct)
+    tmin = _set_tmin(ft_struct)  # create start time
+    info = _create_info(ft_struct)  # create info structure
+
+    custom_epochs = EpochsArray(data=data, info=info, tmin=tmin,
+                                events=events, metadata=metadata)
+    return custom_epochs
+
+
+def read_evoked_fieldtrip(ft_structure_path, comment=None, data_name='data'):
+    """Load evoked data from a FieldTrip timelocked structure.
```

Comment:
ft_structure_path suggests it's a path to a directory. I would just call it `fname` as we do eg here https://github.com/mne-tools/mne-python/blob/master/mne/io/fiff/raw.py#L77

Reply:
all `ft_structure_path` variables were renamed to fname.